### DISCUSSION TITLE
Freedesktop theme icons?

### DISCUSSION THREAD

**Discussion by redtide** [tsujan/Arqiver#28] [Q&A] [answered]
I noticed in (some of) your apps you use a specific monochromatic icon theme, I guess it's the simplest/lightweight way, but I wonder if it could be possible to support also theme icons, though I'm not sure how to solve the case that some are custom, resulting in a mixed collection.

**Comment by tsujan** [comment, marked as answer]
I use hard-coded icons with my apps because:

 1. They're independent of all desktop environments and need icons that may not exist in some well-known icon sets.
 2. I prefer my icons to be clearly visible with dark widget styles too. I can't rely on DEs for it.
 3. Some DEs are Qt-unfriendly and may not show theme icons with Qt apps.

The hard-coded monochrome icons are only for the main window. Dialogs use the theme icons because they're provided by the DE.

In general, what I do may not be good practice but, as long as there are DEs like Gnome that want to be different from the rest of Linux, I have no other option.

**Reply by Ved-un** [reply]
Thanks for the program.
Can you add a setting like in your Featherpad or will it complicate the code too much?


![flameshot_205](https://user-images.githubusercontent.com/48730277/180648180-31b65dd8-32bd-4fb0-89af-186fd2677c98.png)

**Reply by tsujan** [reply]
In FeatherNotes or Arqiver, the most needed icons may not exist in some sets or may have different names.

In FeatherPad, it was possible to take the toolbar and menu icons from the system because they existed in almost all sets. But search and replacement icons are still hard-coded.
